pyautosplit/game.py

```diff
@@ -109,9 +109,19 @@ def get_address(varname):
         for name, var in self.data["variables"].items():
             if var["type"] == "rsp" or var["type"] == "rbp":
                 continue
+
             addr = get_address(name)
             try:
-                if "var_type" in var and var["var_type"] == "bool":
+                if var["type"] == "sptr" or var["type"] == "vptr":
+                    if var["type"] == "sptr":
+                        addr = self.process.base_address + addr
+                    ptr = self.process.read_pointer(addr)
+                    while ptr == 0:
```

Comment:
This seems like a bad idea. What happens if the value at addr is indeed 0? Then this would be an infinite loop

Reply:
My thought was that because other values rely on this being correct. The loop is there because the pointer might change in runtime and is temporary null. If the pointer never points anywhere we're having the wrong memory address and other values are incorrect rendering reading them useless.
Also pyautosplit closes itself right away without a check.

Another option would be to only save the value if it isn't 0. That wouldn't get us stuck in an infinite loop.